packages/react/src/brands/hooks/useBrand.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  fetchBrand as fetchBrandAction,
+  getBrand,
+  getBrandError,
+  isBrandFetched,
+  isBrandLoading,
+  StoreState,
+} from '@farfetch/blackout-redux';
+import { useAction } from '../../helpers';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import type { Brand } from '@farfetch/blackout-client';
+import type { UseBrandOptions } from './types/useBrand';
+
+const useBrand = (id: Brand['id'], options: UseBrandOptions) => {
+  const { enableAutoFetch = true, config } = options;
+
+  const isLoading = useSelector((state: StoreState) =>
+    isBrandLoading(state, id),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isBrandFetched(state, id),
+  );
+  const error = useSelector((state: StoreState) => getBrandError(state, id));
+  const brand = useSelector((state: StoreState) => getBrand(state, id));
+  const fetch = useAction(fetchBrandAction);
+  const fetchBrand = useCallback(() => fetch(id, config), [config, fetch, id]);
+
+  useEffect(() => {
+    if (!isLoading && !error && !isFetched && enableAutoFetch) {
```

Comment:
I know that this should not happen in TS, but I think it would be a good rule of thumb to also check if the id parameter contains a value. It only should not contain a value if the developer forces a cast to `Brand['id']` though.

Also, you can remove the `!error` condition because it is accounted for by the `!isFetched` condition.

Reply:
Yeah, I noticed that `!error `and `!isLoading` it's accounted for by the `!isFetched` condition!  
I only added these 2 conditions to keep the same approach that we had applied for the other hooks.  

I can remove these conditions, but  I would suggest creating a new ticket to refactor the other hooks.